### DISCUSSION TITLE
Can github/gitlab/gogs/etc directly support git-bug issues?

### DISCUSSION THREAD

**Discussion by udhos** [git-bug/git-bug#924] [Q&A] [answered]
Is there any ongoing work to push into github/gitlab/gogs/etc support for handling issues created git-bug ?

It would be awesome.

**Comment by MichaelMure** [comment, marked as answer]
That's part of the "master plan". git-bug is design to be easy to pick-up for them.

However it's probably way too early for such a push. git-bug would need to have a more stable core and data model, and attract actual users.

**Comment by ghost** [comment]
to attract stable users whats the MVP then ?
Being able to use git-bug with github  ? I honestly am not sure but curious the direction you want to head.

**Comment by MichaelMure** [comment]
Actually there is different groups of users and thus different levels of MVP, each being a super-set of the previous.

Here is how I see them and what is required to enable them:



  1. autonomous project (no bugs submitted by non-dev)
    - stable data model (mainly #5)
  1. usage complementary to another tracker, bugs available offline as read-only
    - incremental import from the main bug-tracker (probably github at first)
  1. usage complementary to another tracker, bugs available offline as read/write
    - incremental bidirectional bridge with the main bug-tracker
  1. standalone usage
    - webUI able to serve as a public interface to edit bugs

**Comment by kardianos** [comment]
Out of curiosity, is there a substantial difference between git-bug's internal storage method and NoteDB https://gerrit-review.googlesource.com/Documentation/note-db.html ?

**Comment by MichaelMure** [comment]
From what I gather from [this](https://storage.googleapis.com/gerrit-talks/summit/2015/NoteDB.pdf), it's generally the same idea with some differences. The slides are a bit light in details so it's more an educated guess.

Both git-bug and NoteDb store data in a (linear for git-but, not sure for NoteDb) chain of commit accessible through a ref, don't pollute the normal code workflow and rely on git for storing/transport of the data.

However, NoteDb **seems** to store data as key/value pairs in commit message. git-bug goes [further than that](https://github.com/MichaelMure/git-bug/blob/master/doc/model.md) and store arbitrary data in git Blob and Tree, and tie them in a chain of commits. This gives the sames guarantees but allow for a more flexible format. As an example, git-bug can store files for the embedded media  in comments and have them managed by git as well. This data model also helps to implement the conflict-free merge.

I don't think that Gerrit needs to have that flexibility or merging data, so I'd say it's good enough for them.

**Comment by mcepl** [comment]
See [this discussion on the Sourcehut list](https://lists.sr.ht/~sircmpwn/sr.ht-discuss/%3CCYLGXWF4VOED.3FM8RG6W8JRM5%40cepl.eu%3E).

**Comment by sudoforge** [comment]
I'm going to close this, as it has been answered.
